### ISSUE TITLE
New feature: different thresholds for bike/run

### ISSUE THREAD

**Issue by xanadonf** [thomaschampagne/elevate#331]
You know our thresholds are physiologically different in each sport.
As an example my thresholds are 130/155 on bike and 152/172 while running.
As a consequence -and because I made the choice to set my running values in Strava settings- my Multisports Fitness Trend values are totally wrong.

I asked Strava for this new feature some days ago, while I dont't exactly understand which values are the most significative in Multisports Fitness Trend accuracy :
- Strava TRIMP value?
- Stravistics athlete settings: max and resting heartrates?
I think both.

Considering there are great chances Strava won't implement this functionnality, If this can be set on Stravistix side, it would be a major enhancement for many of us ;o)

**Comment by thomaschampagne** [comment]
TRIMP is calculated from your heartrate streams of your activities and hr max/rest from stravistix settings.

Strava TRIMP (or suffer score) is neved used in Multisports Fitness Trend

Tom

**Comment by xanadonf** [comment]
Thx Tom! So, this seems to be a Stravistix enhancement. 
I think TRIMP should be computed according to HRR (for many of those who do not know their thresholds) or in accord with true thresholds to be more accurate for the others.

**Comment by HeijnenPeter** [comment]
That ties into the things I see, Heart rate is working great for comparing the same activity types, but not between different activity types. I do a lot of boot camp and during the running part I see higher heart rate levels but during pushups, pullups, dips, situps or any other type of core training my heart rate drops to 100-120 which means it is rest, but that never feels that way. Last night I had a very tough workout, but my TRIMP shows only 25. And I do not question the calculation, since it is based on heart rate zones, but comparing it to a rather slow running half an hour, which would give at least 70 in TRIMP, TRIMP is currently not a good measure for bootcamp. Without lactate measurement it will probably never be.....

**Comment by thomaschampagne** [comment]
@xanadonf it's computed from HRR
@HeijnenPeter TRIMP is computed when moving... https://github.com/thomaschampagne/stravistix/blob/develop/plugin/core/scripts/processors/ActivityComputer.ts#L533 Removing this on next update

**Comment by xanadonf** [comment]
Tom, 
HRR based calculation (I understand this is the current way)
OR
true threshold values (my hope).
I mean It would be nice to have the choice according to accuracy we need ;o)

**Comment by thomaschampagne** [comment]
@xanadonf So you would like to have HR min/max per sports?

Tom

**Comment by xanadonf** [comment]
We talked about Multisports Fitness Trend in our triathlon club last week, and for many of (who know their thresholds 1 & 2 values), it it not relevant. Fatigue curve does not correspond to real fatigue (*) and we're regularly in a overload state, which does not correspond to reality.

I made the exercise with a different formula (of my own, which corresponds to strictly nothing physiological) that does not use HRR but threshold values (T1, T2), and my overload periods correspond to real ones next year.

As an example a ride at 140 rpm will correspond to a high TRIMP value (right, my T1 is 130 ppm).
A long run at 140 rpm too (wrong, cause my running T1 is 152, and I'm faaaar below, this is a really resting run).

HR min/max per sports won't be "realistic", but it could make the deal, yes ("en bidouillant" in French :o) )
Calculating a TRIMP (or similar formula / name) would be more accurate.

(*) It will be much worse when swimming activities will be taken into account, I think I'll be dead when I'l see overload values ;o))))
